src/axl.h

```diff
@@ -82,6 +88,23 @@ int AXL_Finalize (void);
         __AXL_Create(type, name, GET_ARG0(__VA_ARGS__))
 int __AXL_Create (axl_xfer_t xtype, const char* name, const char* state_file);
 
+/**
+ * Get/set AXL configuration values.
+ *
+ * config: The new configuration.  Global variables are in top level of
+ *              the tree, and per-ID values are subtrees.  If config=NULL,
+ *              then return a kvtree with all the configuration values (globals
+ *              and all per-ID trees).
+ *
+ * Return value: If config != NULL, then return config on success.  If
+ *                      config=NULL (you're querying the config) then return
+ *                      a new kvtree on success.  Return NULL on any failures.
+ */
+typedef struct kvtree_struct kvtree;
```

Comment:
`kvtree` is already `typedef`'d in kvtree.h.

Reply:
The point here, as requested earlier in this ticket, is to not require user to include `kvtree.h` is they are not making use of `AXL_Config` hence I must forward declare the type. DId the `#include "kvtree.h"` line creep back in.

Reply:
Ah you're right, I assumed you had `#include <kvtree.h>` in `axl.h` for the `struct kvtree_struct` definition, but the way you typedefed it gets around needing that definition unless you use it.  Very cool trick! :+1: